src/app/pages/HomePage/Graph/NetworkSelector/index.tsx

```diff
@@ -119,12 +120,19 @@ const NetworkSelectorButton = forwardRef(
 interface NetworkSelectProps {
   network: Network
   setNetwork: (network: Network | null) => void
+  forLayer?: Layer
 }
 
-export const NetworkSelector: FC<NetworkSelectProps> = ({ network, setNetwork }) => {
+export const NetworkSelector: FC<NetworkSelectProps> = ({ network, setNetwork, forLayer }) => {
   const { t } = useTranslation()
 
-  const options = RouteUtils.getEnabledNetworks().map(network => ({
+  const networks = forLayer
+    ? RouteUtils.getEnabledNetworksForLayer(forLayer)
+    : RouteUtils.getEnabledNetworks()
```

Comment:
I have seen this piece of code multiple times now, should be moved to utils I guess.

Reply:
Done